src/test/java/uk/gov/companieshouse/web/pps/validation/EnterDetailsValidatorTest.java

```diff
@@ -32,125 +31,71 @@ void setUp() {
-        assertEquals("Enter your penalty reference exactly as shown on your penalty letter",
-                Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
-    }
-
-    @Test
-    void isValidWhenRefStartsWithSanctions() {
-        enterDetails.setPenaltyReferenceName(SANCTIONS.name());
-        enterDetails.setPenaltyRef("P1234567");
-        enterDetails.setCompanyNumber("12345678");
+    @ParameterizedTest
+    @CsvSource({
+            "AB123456,A1234567,LATE_FILING",
+            "AB123456,P1234567,SANCTIONS",
+            "OE123456,U1234567,SANCTIONS_ROE"
+    })
+    void isValidWhenRefStart(String companyId, String penaltyRef, String referenceName) {
+        enterDetails.setPenaltyReferenceName(referenceName);
+        enterDetails.setPenaltyRef(penaltyRef);
+        enterDetails.setCompanyNumber(companyId);
         BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
 
         testValidator.isValid(enterDetails, bindingResult);
 
         assertFalse(bindingResult.hasErrors());
     }
 
-    @Test
-    void isNotValidWhenPenaltyRefIsMissingForSanctions() {
-        enterDetails.setPenaltyReferenceName(SANCTIONS.name());
-        enterDetails.setPenaltyRef("");
-        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
-
-        testValidator.isValid(enterDetails, bindingResult);
-
-        assertTrue(bindingResult.hasErrors());
-        assertEquals(1, bindingResult.getFieldErrorCount(PENALTY_REF_FIELD));
-        assertEquals("Enter the penalty reference",
-                Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
-    }
-
-    @Test
-    void isNotValidWhenPenaltyRefIsNotValidForSanctions() {
-        enterDetails.setPenaltyReferenceName(SANCTIONS.name());
-        enterDetails.setPenaltyRef("PN1234567");
-        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
-
-        testValidator.isValid(enterDetails, bindingResult);
-
-        assertTrue(bindingResult.hasErrors());
-        assertEquals(1, bindingResult.getFieldErrorCount(PENALTY_REF_FIELD));
-        assertEquals("Enter your penalty reference exactly as shown on your penalty letter",
-                Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
-    }
-
-    @Test
-    void isNotValidWhenCompanyNumberContainsSpaces() {
+    @ParameterizedTest
+    @CsvSource({
+            "AB12345,Company number must be 8 characters",
+            "AB12345!,Company number must only contain numbers and letters",
+            " ,Enter the company number",
+            "X12345678, Enter the company number exactly as shown on your penalty notice"
+    })
+    void isNotValidForLateFilingCompanyIdCheck(String companyId, String errorMessage) {
```

Comment:
Similar to previous comment, the parameter should be companyNumber and name should be CompanyNumberCheck?

```suggestion
    void isNotValidForLateFilingCompanyNumberCheck(String companyNumber, String errorMessage) {
```